plugins/xml-adapter/src/main/java/org/polypheny/db/adapter/xml/XmlToPolyConverter.java

```diff
@@ -0,0 +1,124 @@
+                break; // Stop processing at the end of the current document element
+            }
+        }
+        return PolyMap.of( map );
+    }
+
+
+    public PolyValue nodeToPolyValue( XMLStreamReader reader ) throws XMLStreamException {
+        if ( reader.getEventType() == XMLStreamConstants.START_ELEMENT ) {
+            String type = reader.getAttributeValue( null, "type" );
+            return convertByType( reader, type != null ? type : "string" );
+        }
+        return new PolyNull(); // Default case for unsupported nodes
+    }
+
+
+    private PolyValue convertByType( XMLStreamReader reader, String type ) throws XMLStreamException {
+        StringBuilder value = new StringBuilder();
+        while ( reader.hasNext() ) {
+            int event = reader.next();
+            if ( event == XMLStreamConstants.CHARACTERS ) {
+                value.append( reader.getText().trim() );
+            } else if ( event == XMLStreamConstants.END_ELEMENT ) {
+                break;
+            }
+        }
+
+        if ( value.length() == 0 ) {
+            return new PolyNull();
+        }
+
+        String valueStr = value.toString();
+        return switch ( type ) {
+            case "boolean" -> new PolyBoolean( Boolean.parseBoolean( valueStr ) );
+            case "integer" -> new PolyLong( Long.parseLong( valueStr ) );
+            case "decimal" -> new PolyBigDecimal( new java.math.BigDecimal( valueStr ) );
+            case "float" -> new PolyFloat( Float.parseFloat( valueStr ) );
+            case "double" -> new PolyDouble( Double.parseDouble( valueStr ) );
+            case "date" -> {
+                LocalDate date = LocalDate.parse( valueStr, DateTimeFormatter.ISO_DATE );
+                yield new PolyDate( date.atStartOfDay().toInstant( java.time.ZoneOffset.UTC ).toEpochMilli() );
+            }
+            case "time" -> {
+                LocalTime time = LocalTime.parse( valueStr, DateTimeFormatter.ISO_TIME );
+                yield new PolyTime( (int) (time.toNanoOfDay() / 1000000) );
+            }
+            case "dateTime" -> {
+                LocalDateTime dateTime = LocalDateTime.parse( valueStr, DateTimeFormatter.ISO_DATE_TIME );
+                yield new PolyTimestamp( dateTime.toInstant( java.time.ZoneOffset.UTC ).toEpochMilli() );
+            }
+            case "base64Binary" -> {
+                byte[] binaryData = Base64.getDecoder().decode( valueStr );
+                yield new PolyBinary( binaryData, binaryData.length );
+            }
+            case "hexBinary" -> {
+                try {
+                    byte[] hexBinaryData = Hex.decodeHex( valueStr.toCharArray() );
+                    yield new PolyBinary( hexBinaryData, hexBinaryData.length );
+                } catch ( Exception e ) {
+                    yield new PolyString( valueStr ); // Fallback in case of error
```

Comment:
Why is `hexBinary` not as strict as the others?

Reply:
The decoder exception is not propagated  up to the moveNext in the enumerator where an exception is thrown.